### ISSUE TITLE
Hasher output range

### ISSUE THREAD

**Issue by snappizz** [supercollider/supercollider#2588]
The docs say that 

> returns a unique output value from zero to one for each input value according to a hash function.

But running `Hasher.ar(Sweep.ar)`, I can only get values in the range (-0.2, 0.2). Would it be better to fix the range or to document the quirk?

**Comment by brianlheim** [comment]
I can't recreate this—if I run 
```
{Hasher.ar(Sweep.ar) }.plot(0.01)
```
the first five values I get are
```
0.89006209373474, 0.85798931121826, 0.95394110679626, -0.15698313713074, -0.87864685058594
```

I'm confused as to why this UGen doesn't give constant output for constant input. Even though the doc says this is normal, it doesn't seem to follow from the idea that it's based on a hash function. For example, I get this:
`Hasher.ar(0)` -> `-0.45145511627197, 0.50862240791321, -0.45145511627197, -0.45145511627197, -0.45145511627197, -0.82332158088684` ...

I did some digging in the source code and the hash function being used is deterministic. I'm probably missing something obvious here since I'm not that familiar with the details of the server code.

**Comment by snappizz** [comment]
Well, that's weird, I swore I got ±0.2 but I can't reproduce it now. Still, the documentation is wrong, it should say -1 to +1 and not 0 to 1.

> I'm confused as to why this UGen doesn't give constant output for constant input.

That's because `0` is a control-rate input and you're calling Hasher at audio rate. Hasher thinks it has an audio-rate input, so it's reading garbage. This is a bug caused by Hasher not checking the rate of its inputs.

**Comment by brianlheim** [comment]
> That's because 0 is a control-rate input and you're calling Hasher at audio rate.

Oh duh. `DC.ar(0)` as input works fine.

**Comment by telephon** [comment]
closed via  #2589
